File: src/app/models/Chauffeur.ts
import { MandatRecieve } from "./MandatRecieve";
import { SocieteModel } from "./Societe";

export interface chauffeur{
    nom:string,
    cin:string,
    adresse:string;
    photo:string;
    mandat:MandatRecieve,
    societe:SocieteModel,

}

File: src/main/java/org/sid/stage_prj/entites/Chauffeur.java
package org.sid.stage_prj.entites;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Chauffeur {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nom;
    private String prenom;
    private String cin;
    private String adresse;
    private String photo;
    @OneToOne
    private Mandat mandat;

    @ManyToOne
    private Societe societe;

    @OneToOne(mappedBy = "chauffeur")
    private Camion camion;
}

